### PR TITLE
Commerce 9716

### PR DESCRIPTION
**Pull request by MahmoudAzzam**
https://issues.liferay.com/browse/COMMERCE-9716

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Ehi @MahmoudAzzam, please

1. move the autogenerated commits to the end of the PR
2. add the field also to the DTO converter of the order item in the API
3. See CI results, looks like you need to run also baseLine command

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/service.xml [hunk 1]
```diff
@@ -357,6 +357,7 @@
 		<column name="json" type="String" />
 		<column name="manuallyAdjusted" type="boolean" />
 		<column name="discountManuallyAdjusted" type="boolean" />
+		<column name="priceManuallyAdjusted" type="boolean" />
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/service.xml:360]: I think this should be alphabetically sorted, also the discountManuallyAdjusted property, I'm sorry I didn't catch it before

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/registry/CommerceServiceUpgradeStepRegistrator.java [hunk 2]
```diff
@@ -403,6 +403,11 @@ protected String[][] getTableAndPrimaryKeyColumnNames() {
 			UpgradeProcessFactory.addColumns(
 				"CommerceOrderItem", "discountManuallyAdjusted BOOLEAN"));
 
+		registry.register(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/registry/CommerceServiceUpgradeStepRegistrator.java:406]: I think you can do only one UP if we send the two tickets together

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/rest-openapi.yaml [hunk 3]
```diff
@@ -727,6 +727,9 @@ components:
                 printedNote:
                     example: Order item printed note
                     type: string
+                priceManuallyAdjusted:
+                    example: true
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/rest-openapi.yaml:731]: add the readonly property

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/rest-openapi.yaml:731]: also sort alphabetically

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/dto/v1_0/converter/OrderItemDTOConverter.java [hunk 4]
```diff
@@ -108,6 +108,7 @@ public OrderItem toDTO(DTOConverterContext dtoConverterContext)
 					commerceOrder.getExternalReferenceCode();
 				orderId = commerceOrder.getCommerceOrderId();
 				printedNote = commerceOrderItem.getPrintedNote();
+				priceManuallyAdjusted = commerceOrderItem.getPriceManuallyAdjusted();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/dto/v1_0/converter/OrderItemDTOConverter.java:111]: sort alphabetically

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Ehi @MahmoudAzzam overall this is good, but there are some little changes that you need to do before sending this to Brian.

Thanks!

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java [hunk 5]
```diff
@@ -1566,6 +1586,67 @@ private List<CommerceOptionValue> _getStaticOptionValuesNotLinkedToSku(
...
+
+	private boolean _isDiscountChanged(
+		BigDecimal discountAmount, BigDecimal discountPercentageLevel1,
+		BigDecimal discountPercentageLevel2,
+		BigDecimal discountPercentageLevel3,
+		BigDecimal discountPercentageLevel4,
+		CommerceOrderItem commerceOrderItem) {
+
+		if ((commerceOrderItem.getDiscountAmount() != discountAmount) ||
+			(commerceOrderItem.getDiscountPercentageLevel1() !=
```
[the remark this answers is not in the document]
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java:1649]: The AC says discount amount, also if you change the percentage and this flag prevents the auto-recalculate the amount will not change, meaning you are not really changing the discount.
That said, I think we should not check for percentage level changes.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/dto/v1_0/converter/OrderItemDTOConverter.java (hunk 4 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/dto/v1_0/converter/OrderItemDTOConverter.java:111]: CI is also angry about this  :)

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]
@MahmoudAzzam fix commit names, pretty much all of them are about discountManuallyAdjusted but you are also working with priceManuallyAdjusted from the ticket COMMERCE-9784, so it is unclear what the commits are really about.

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]

**Review by luca-pellizzon — COMMENTED** [review]
@MahmoudAzzam probably my comment was not clear, you can put commits togheter, I just want a meaninfull message, so that I know what I will find inside.
That said, here you also split autogenerated code commits, not sure this is fine to Brian even if we are talking about different jira ticket.
@alessiorendina @andrea-ale-sbarra  what do you think?

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
@MahmoudAzzam I still see rebasing issues. Could please take a look? Thanks.

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
@MahmoudAzzam please try to rebase this with upstream/master branch and relaunch the tests. Thank you!

**Comment by MahmoudAzzam** [comment]
new PR created @ https://github.com/liferay-commerce/liferay-portal/pull/2985

**Pull request closed without merging**